### DISCUSSION TITLE
fabric_rpc:map_view/5 exit:timeout

### DISCUSSION THREAD

**Discussion by ivailopp** [apache/couchdb#3016] [Q&A]
Hi We have timeout error massage in log .

[error] 2020-07-20T17:35:51.461866Z couchdb@127.0.0.1 <0.20056.2805> 9654b6ead9 rexi_server: from: couchdb@127.0.0.1(<0.15410.2691>) mfa: fabric_rpc:map_view/5 exit:timeout [{rexi,init_stream,1,[{file,"src/rexi.erl"},{line,290}]},{rexi,stream2,3,[{file,"src/rexi.erl"},{line,230}]},{fabric_rpc,view_cb,2,[{file,"src/fabric_rpc.erl"},{line,458}]},{couch_mrview,map
_fold,3,[{file,"src/couch_mrview.erl"},{line,469}]},{couch_mrview_util,fold_fun,4,[{file,"src/couch_mrview_util.erl"},{line,374}]},{couch_btree,stream_kv_node2,8,[{file,"src/couch_btree.erl"},{line,851}]},{couch_btree,stream_kp_node,8,[{file,"src/couch_btree.erl"},{line,822}]},{couch_btree,fold,4,[{file,"src/couch_btree.erl"},{line,224}]}]
[error] 2020-07-20T17:35:51.514804Z couchdb@127.0.0.1 <0.19770.2805> 6840e8451e rexi_server: from: couchdb@127.0.0.1(<0.17711.2805>) mfa: fabric_rpc:map_**view/5 exit:timeout [**{rexi,init_stream,1,[{file,"src/rexi.erl"},{line,290}]},{rexi,stream2,3,[{file,"src/rexi.erl"},{line,230}]},{fabric_rpc,view_cb,2,[{file,"src/fabric_rpc.erl"},{line,458}]},{couch_mrview,fin
ish_fold,2,[{file,"src/couch_mrview.erl"},{line,627}]},{couch_mrview,query_view,5,[{file,"src/couch_mrview.erl"},{line,281}]},{rexi_server,init_p,3,[{file,"src/rexi_server.erl"},{line,138}]}]

[NOTE]: # ( ^^ Provide a general summary of the issue in the title above. ^^ )

## Description

I saw error in our log - connections timeout .
mfa: fabric_rpc:map_view/5 exit:timeout [
How i can fix this ?

Is this part of fabric?

This is my config.
default.ini

; [fabric]
; all_docs_concurrency = 10
; changes_duration =
; shard_timeout_factor = 2
; uuid_prefix_len = 7
; request_timeout = 60000
; all_docs_timeout = 10000
; attachments_timeout = 60000
; view_timeout = 3600000
; partition_view_timeout = 3600000

## Your Environment

CouchDB 3.1
RHEL 7.6

Any help would be greatly appreciated.

Thnaks

**Comment by janl** [comment]
Fabric timeouts happen when nodes in a cluster take too long to reply to requests from other nodes. Those timeouts can stem from multiple sources: slow network, overloaded CPU, overloaded disk IO, etc.

It looks like you are running a single node setup (as evidenced by your node name couchdb@127.0.0.1), in this case, fanric_rpc will still run, but the „other“ node is just the same node. That said, the error conditions are the same as above.

**Reply by baversjo** [reply]
@janl what value would you tweak for increasing the fabric timeout value? And, would you recommend changing it in a clustered setup?
